repos/system_upgrade/el8toel9/actors/checkblsgrubcfgonppc64/libraries/blsgrubcfgonppc64.py

```diff
@@ -0,0 +1,36 @@
+from leapp import reporting
+from leapp.libraries.common import grub
+from leapp.libraries.stdlib import api
+from leapp.models import DefaultGrubInfo, GrubCfg
+
+
+def _is_system_ppc64():
+
+    arch = api.current_actor().configuration.architecture
+    return True if arch == 'ppc64le' else False
+
+
+def process():
+
+    default_grub_msg = next(api.consume(DefaultGrubInfo), None)
+    grub_cfg = next(api.consume(GrubCfg), None)
+
+    if (
+        not grub.is_grub_cfg_bls(grub_cfg) and
+        _is_system_ppc64() and
+        grub.is_blscfg_enabled(default_grub_msg)
+    ):
+        reporting.create_report([
+            reporting.Title(
+                'Leapp will execute "grub2-mkconfig" to fix BLS Grub configuration.'
+            ),
+            reporting.Summary(
+                'On "ppc64le" systems with BLS enabled, the GRUB configuration is not '
+                'properly converted after the upgrade and Leapp has to run "grub2-mkconfig" '
+                '-o /boot/grub2/grub.cfg command in order to fix an issue with booting into '
+                'the RHEL 8 kernel instead of RHEL 9.'
+
+            ),
+            reporting.Severity(reporting.Severity.HIGH),
+            reporting.Tags([reporting.Tags.BOOT]),
+        ])
```

Comment:
from my understanding, is it possible to properly update the bootloader configuration on RHEL 8 system from user POV? I can imagine users could be afraid when they read the msg :) We could give them a remediation instructions how they can fix their bootloader configuration prior the upgrade on their own, so we will not do it. wdyt?

Reply:
discussed, there is nothing what could be done before the upgrade because of bug in grub (the actions is not possible to do before the new grub is installed).